### ISSUE TITLE
get_visible_pages in menu.py doesn't scale

### ISSUE THREAD

**Issue by xmedia-systems** [divio/django-cms#1206]
I've imported some 1000 pages into django-cms and got a "Too many SQL variables" exception from sqlite. The problem is get_visible_pages in menu.py. It adds  all page ids in the cms as a complicated condition to the where clause. Of course, this doesn't scale at all! I'm going to think about a better solution but for the time being: woldn't it be a good idea to omit permissions checking if CMS_PERMISSION is set to False? i.e.:

    if not settings.CMS_PERMISSION or (request.user.is_staff and settings.CMS_PUBLIC_FOR in ('staff', 'all')):
        return [page.pk for page in pages]

**Comment by piotrjakimiak** [comment]
Now problem is in `load_view_restrictions`. It make query with `__in` lookup, which fail for large amount of pages in sql.
Trying to fix it will lead to increasing number of queries for other databases or creating special cases for sqlite.

As far as i've tested `get_visible_pages` works fast for large data (miliseconds for about 500 pages). I've simplified that method in #5115 but it doesn't solve sqlite problem.

What's more in lookup is used in more places so sqlit will probably fail in other place when using with large data.

**Comment by czpython** [comment]
Closing this as a won't fix.
We've made several performance improvements in 3.4 in regards to the page tree loading time on the admin.
That said, I know for a fact we're still adding the page ids as look ups in the SQL query and as such this is not fully fixed but as pointed out above, changing this would slow things down for other databases.

Feel free to create a new ticket if this is causing problems outside of SQLite.
